src/components/Editor/index.js

```diff
@@ -220,6 +222,31 @@ class Editor extends Component {
     shortcuts.off(`CmdOrCtrl+${searchAgainKey}`);
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
+    if (Object.keys(nextProps).length != Object.keys(this.props).length) {
+      return true;
+    }
+    if (Object.keys(nextState).length != Object.keys(this.state).length) {
+      return true;
+    }
+
+    let key;
+
+    for (key in nextProps) {
+      if (key != "getExpression" && nextProps[key] != this.props[key]) {
+        return true;
+      }
+    }
+
+    for (key in nextState) {
+      if (nextProps[key] != this.props[key]) {
+        return true;
+      }
+    }
```

Comment:
generally speaking this looks like we can extract this to a `utils/react.js` file that takes props, state, and a blacklist for props and state.

I'm also okay merging this soon and then removing it with selectors. This way we have a short-term fix.